Typescript: ThisType in ActionHandler is not work

## Prerequisites

Please answer the following questions for yourself before submitting an issue. 

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I'm reporting the issue to the correct repository

## Current Behavior

<!-- What is the current behavior? -->
For example code below:
```typescript
    this.parseServiceSchema({
      name: 'user',
      mixins: [DbService],
      actions: {
        login: {
          rest: 'POST /login',
          async handler(ctx: Context) {
            return {
              id: 1,
              username: 'test',
            };
          },
        },
      },
    });
```

I try to typing `this.` and not any type suggest display.

When i try to check the type of `this`, just any.

## Expected Behavior

<!-- Please describe the behavior you are expecting -->

Maybe we should change 
```typescript
type ActionHandler<T = any> = ((ctx: Context<any, any>) => Promise<T> | T) & ThisType<Service>;
```
to
```typescript
type ActionHandler<T = any> = ((this: Service, ctx: Context<any, any>) => Promise<T> | T);
```

`ThisType` looks like use in parent, not in `ActionHandler` self

https://github.com/moleculerjs/moleculer/blob/7b94f7cac722f5f82a1b5ed8906a915fa586b79f/index.d.ts#L47

Hi, 

I'm not sure if this is related, but i cannot use this with typescript in the following snippet : 

```ts
   public constructor(public broker: ServiceBroker) {
        super(broker);
        this.parseServiceSchema({
            name: "greeter",
            actions:{
                hello: {
                    async handler(): Promise<string> {
                        return this.ActionHello();
                    },
                },
                welcome: {
                    
                    params: {
                        name: "string",
                    },
                    async handler(ctx: Context<{name: string}>): Promise<string> {
                        await this.actions.hello(null, {parentCtx: ctx})
                        return this.ActionWelcome(ctx.params.name);
                    },
                },
            },
        });
    }
...
}
```

This file does not compile with the following error from ts : 

```
[2021-07-02T15:00:18.889Z] ERROR citopia-theo.local-30460/BROKER: Failed to load service '/Users/dprslt/Projects/moleculer/node1/src/services/greeter.service.ts' src/services/greeter.service.ts:24:31 - error TS2684: The 'this' context of type 'ServiceActions' is not assignable to method's 'this' of type 'Service<ServiceSettingSchema>'.
  Type 'ServiceActions' is missing the following properties from type 'Service<ServiceSettingSchema>': parseServiceSchema, name, fullName, settings, and 12 more.

24                         await this.actions.hello(null, {parentCtx: ctx})
                                 ~~~~~~~~~~~~

```

I've tried with a simple export, but the problem stay the same

```ts
const ImagesService: ServiceSchema = {
    name: "files.images",
        actions:{
            ... 
        }
}
```

This error is only ts related because a `@ts-ignore` make it works.

`"version": "0.14.14"`

--- a/index.d.ts
+++ b/index.d.ts
@@ -44,10 +44,7 @@ declare namespace Moleculer {
 		trace(...args: any[]): void;
 	}
 
-	type ActionHandler<T = any, S = ServiceSettingSchema> = (
-		this: Service<S>,
-		ctx: Context<any, any>
-	) => Promise<T> | T;
+	type ActionHandler<T = any> = (ctx: Context<any, any>) => Promise<T> | T;
 	type ActionParamSchema = { [key: string]: any };
 	type ActionParamTypes =
 		| "any"
@@ -460,7 +457,7 @@ declare namespace Moleculer {
 		basePath?: string,
 	}
 
-	interface ActionSchema {
+	type ActionSchema<S = ServiceSettingSchema> = {
 		name?: string;
 		rest?: RestSchema | string | string[],
 		visibility?: ActionVisibility;
@@ -476,9 +473,9 @@ declare namespace Moleculer {
 		hooks?: ActionHooks;
 
 		[key: string]: any;
-	}
+	} & ThisType<Service<S>>
 
-	interface EventSchema {
+	type EventSchema<S = ServiceSettingSchema> = {
 		name?: string;
 		group?: string;
 		params?: ActionParams;
@@ -489,9 +486,11 @@ declare namespace Moleculer {
 		context?: boolean;
 
 		[key: string]: any;
-	}
+	} & ThisType<Service<S>>
 
-	type ServiceActionsSchema = { [key: string]: ActionSchema | ActionHandler | boolean; };
+	type ServiceActionsSchema<S = ServiceSettingSchema> = {
+		[key: string]: ActionSchema | ActionHandler | boolean;
+	} & ThisType<Service<S>>;
 
 	class BrokerNode {
 		id: string;
@@ -599,17 +598,11 @@ declare namespace Moleculer {
 		[name: string]: any;
 	}
 
-	type ServiceEventLegacyHandler = (
-		this: Service,
-		payload: any,
-		sender: string,
-		eventName: string,
-		ctx: Context
-	) => void;
+	type ServiceEventLegacyHandler = (payload: any,	sender: string,	eventName: string, ctx: Context) => void;
 
-	type ServiceEventHandler = (this: Service, ctx: Context) => void;
+	type ServiceEventHandler = (ctx: Context) => void;
 
-	interface ServiceEvent {
+	type ServiceEvent<S = ServiceSettingSchema> = {
 		name?: string;
 		group?: string;
 		params?: ActionParams;
@@ -617,16 +610,11 @@ declare namespace Moleculer {
 		debounce?: number;
 		throttle?: number;
 		handler?: ServiceEventHandler | ServiceEventLegacyHandler;
-	}
+	} & ThisType<Service<S>>
 
 	type ServiceEvents = { [key: string]: ServiceEventHandler | ServiceEventLegacyHandler | ServiceEvent };
 
-	type ServiceMethods = {
-		[key: string]: <S = ServiceSettingSchema>(
-			this: Service,
-			...args: any[]
-		) => any;
-	};
+	type ServiceMethods = {	[key: string]: (...args: any[]) => any } & ThisType<Service>;
 
 	type CallMiddlewareHandler = (actionName: string, params: any, opts: CallingOptions) => Promise<any>;
 	type Middleware = {
@@ -639,10 +627,7 @@ declare namespace Moleculer {
 			| ((handler: CallMiddlewareHandler) => CallMiddlewareHandler)
 	}
 
-	type MiddlewareInit = (
-		this: ServiceBroker,
-		broker: ServiceBroker
-	) => Middleware;
+	type MiddlewareInit = (broker: ServiceBroker) => Middleware;
 	interface MiddlewareCallHandlerOptions {
 		reverse?: boolean
 	}
@@ -700,13 +685,7 @@ declare namespace Moleculer {
 		[name: string]: any;
 	}
 
-	type ServiceAction = <
-		T = Promise<any>,
-		P extends GenericObject = GenericObject,
-	>(
-		params?: P,
-		opts?: CallingOptions
-	) => T;
+	type ServiceAction = <T = Promise<any>,	P extends GenericObject = GenericObject>(params?: P, opts?: CallingOptions) => T;
 
 	interface ServiceActions {
 		[name: string]: ServiceAction;

--- a/package.json
+++ b/package.json
@@ -220,7 +220,10 @@
   },
   "types": "./index.d.ts",
   "tsd": {
-    "directory": "test/typescript/tsd"
+    "directory": "test/typescript/tsd",
+    "compilerOptions": {
+      "noImplicitThis": true
+    }
   },
   "jest": {
     "coverageDirectory": "../coverage",

